### ISSUE TITLE
namedPlaceholders and array

### ISSUE THREAD

**Issue by sbscan** [mariadb-corporation/mariadb-connector-nodejs#248]
When using namedPlaceholders if we use array parameters it sterilize array without quotes.
My parameter is an array like role => ["Admin","Accounting"]
When I use it as a namedPlaceholder in an insert statement the resulting SQL is: 'role': Admin,Teklif
Which should be: 'role': 'Admin,Teklif'
So ends with an SQL error.
Quick fix is using toString and convert array to string.

**Comment by rusher** [comment]
That's the expected behavior : array permits sending multiple values. 
Example : 
```javascript
    await conn.query('CREATE TABLE namedPlaceholders1(t varchar(128))');
    await conn.query("INSERT INTO `namedPlaceholders1` value ('a'), ('b'), ('c')");
    const res = await conn.query('select * from `namedPlaceholders1` where t IN (:possible)', { possible: ['a', 'c'] });`
   // res = [{ t: 'a' }, { t: 'c' }]
```
if your behavior is explicitly to have a string list, then, yes toString is the way to go.

**Comment by mikerowe81** [comment]
I agree this is desired when the array placeholder is in a `WHERE` clause with the `IN` operator, however, when the array placeholder is an `INSERT` or `UPDATE` value for a JSON column, I expected the value to be `[1,2,3]` rather than `1,2,3`

Would it be possible to check if the placeholder is in the WHERE clause or surrounded by parentheses and use `[].toString()` otherwise use `JSON.stringify([])`
